catroid/src/org/catrobat/catroid/ui/fragment/SpritesListFragment.java

```diff
@@ -358,6 +386,30 @@ public void startDeleteActionMode() {
 		}
 	}
 
+	private void moveSpriteDown() {
+		Collections.swap(spriteList, spritePosition + 1, spritePosition);
+		spriteAdapter.notifyDataSetChanged();
+	}
+
+	private void moveSpriteUp() {
+		Collections.swap(spriteList, spritePosition - 1, spritePosition);
+		spriteAdapter.notifyDataSetChanged();
+	}
+
+	private void moveSpriteBottom() {
+		for (int i = spritePosition; i < spriteList.size() - 1; i++) {
+			Collections.swap(spriteList, i, i + 1);
+		}
```

Comment:
Use [http://developer.android.com/reference/java/util/Collections.html#rotate(java.util.List<?>, int)](http://developer.android.com/reference/java/util/Collections.html#rotate(java.util.List<?>, int)) in both cases, top and bottom.

moveSprite**To**Bottom and moveSprite**To**Top